sql/core/src/main/scala/org/apache/spark/sql/execution/streaming/ValueStateImpl.scala

```diff
@@ -32,11 +34,14 @@ import org.apache.spark.sql.types._
  * variables used in the streaming transformWithState operator.
  * @param store - reference to the StateStore instance to be used for storing state
  * @param stateName - name of logical state partition
+ * @param keyEnc - Spark SQL encoder for key
+ * @tparam K - data type of key
  * @tparam S - data type of object that will be stored
  */
-class ValueStateImpl[S](
+class ValueStateImpl[K, S](
```

Comment:
My intuition is that value can be serialized with Spark SQL, but again you can file a JIRA ticket for this as well. (But that ticket should be a blocker on releasing this API.) We don't release the API as it is, so it's OK.

Reply:
Same here. Created JIRA - https://issues.apache.org/jira/browse/SPARK-46853